sktime/forecasting/model_evaluation/_functions.py

```diff
@@ -137,40 +196,44 @@ def _evaluate_window(
             forecaster.update(y_train, X_train, update_params=update_params)
         fit_time = time.perf_counter() - start_fit
 
+        # predict based on metrics
         pred_type = {
             "pred_quantiles": "predict_quantiles",
             "pred_interval": "predict_interval",
             "pred_proba": "predict_proba",
-            None: "predict",
+            "pred": "predict",
         }
-        # predict
-        start_pred = time.perf_counter()
-
-        if hasattr(scoring, "metric_args"):
-            metric_args = scoring.metric_args
-        else:
-            metric_args = {}
-
-        if hasattr(scoring, "get_tag"):
-            scitype = scoring.get_tag("scitype:y_pred", raise_error=False)
-        else:
-            # If no scitype exists then metric is not proba and no args needed
-            scitype = None
-
-        methodname = pred_type[scitype]
-        method = getattr(forecaster, methodname)
-
-        y_pred = method(fh, X_test, **metric_args)
-        pred_time = time.perf_counter() - start_pred
-        # score
-        score = scoring(y_test, y_pred, y_train=y_train)
+        # cache prediction from the first scitype and reuse it to compute other metrics
+        for scitype in scoring:
+            cache_y_pred = pd.NA
+            for idx, metric in enumerate(scoring.get(scitype)):
+                if idx == 0:
```

Comment:
I think it would be both cleaner in logic and usage if you precompute all the predictions required?

Reply:
To avoid repeating the loop twice - once for precomputing all predictions and once for metrics of each scitype 

I precompute one scitype at a time, and then compute the metrics afterwards in a looping fashion. I think this is cleaner than having the if statement in the inner loop, right?

Reply:
yes, that also works. Perhaps simpler.